nix/master.nix

```diff
@@ -391,98 +399,77 @@ in
-                )"
-                },
-                gitea=${
-                  if !cfg.gitea.enable then
-                    "None"
-                  else
-                    "GiteaConfig(
-                    instance_url=${builtins.toJSON cfg.gitea.instanceUrl},
-                    oauth_id=${builtins.toJSON cfg.gitea.oauthId},
-                    topic=${builtins.toJSON cfg.gitea.topic},
-                )"
-                },
-                build_retries=${builtins.toJSON cfg.buildRetries},
-                cachix=${
-                  if cfg.cachix.name == null then
-                    "None"
-                  else
-                    "CachixConfig(
-                    name=${builtins.toJSON cfg.cachix.name},
-                    signing_key_secret_name=${
-                                        if cfg.cachix.signingKeyFile != null then builtins.toJSON "cachix-signing-key" else "None"
-                                      },
-                    auth_token_secret_name=${
-                                        if cfg.cachix.authTokenFile != null then builtins.toJSON "cachix-auth-token" else "None"
-                                      },
-                )"
-                },
-                admins=${builtins.toJSON cfg.admins},
-                url=${builtins.toJSON config.services.buildbot-nix.master.webhookBaseUrl},
-                nix_eval_max_memory_size=${builtins.toJSON cfg.evalMaxMemorySize},
-                nix_eval_worker_count=${
-                  if cfg.evalWorkerCount == null then "None" else builtins.toString cfg.evalWorkerCount
-                },
-                nix_supported_systems=${builtins.toJSON cfg.buildSystems},
-                outputs_path=${if cfg.outputsPath == null then "None" else builtins.toJSON cfg.outputsPath},
-                post_build_steps=[
-                  ${lib.concatMapStringsSep ",\n" ({ name, environment, command }: ''
-                    steps.ShellCommand(
-                      name=${builtins.toJSON name},
-                      env={
-                        ${lib.concatMapStringsSep ",\n" ({name, value}: ''
-                          ${name}: ${interpolateToString value}
-                        '') (lib.mapAttrsToList lib.nameValuePair environment)}
-                      },
-                      command=[
-                        ${lib.concatMapStringsSep ",\n" (value:
-                          interpolateToString value
-                        ) (if lib.isList command then command else [ command ])}
-                      ]
-                    )
-                  '') cfg.postBuildSteps}
-                ]
+              BuildbotNixConfig.model_validate(json.loads(Path("${(pkgs.formats.json {}).generate "buildbot-nix-config.json" {
+                db_url = cfg.dbUrl;
+                auth_backend = cfg.authBackend;
+                build_retries = cfg.buildRetries;
+                cachix = if !cfg.cachix.enable then
+                  null
```

Comment:
Indentation looks super odd here. I think, we should switch to nixfmt.

Reply:
left that for another PR, I say we commit this eye sore and figure out nixfmt next (or alejandra which i use personally)

Reply:
yes.